[Blake Tan Ting Yu] iP

blaketingyu opened the pull request:
# **Taskmaster** :skull:
>"Happiness is nothing more than good health and a bad memory" – Albert Schweitzer 

Bad memory? Taskmaster will help you with recording your tasks!
* text-based 
* Super easy to learn
* For all ages
* ~~FAST~~ *Super blazing Fast* to use


Getting Started:
1. Download it [here](https://github.com/blaketingyu/ip/releases/tag/A-Jar). 
2. Double click the icon.
3. Add your tasks. 
4. Taskmaster will handle all your annoying tasks. :exploding_head:

Most importantly, it is **FREE?!**
Features:
- [X] Managing Tasks
- [X] Managing Deadlines
- [ ] Reminders (Coming soon) 


Even if you are not a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
 public static void main(String[] args) {
        Taskmaster taskMaster = new Taskmaster();
        taskMaster.ui.runChatBot();
    }

```

Commit:
Add Gradle support

blaketingyu committed:
Level-1

blaketingyu committed:
Level-2

blaketingyu committed:
Level-3

blaketingyu committed:
Level-4

blaketingyu committed:
Level-5

blaketingyu committed:
Level-6

blaketingyu committed:
Moved all the if else cases to Commands file

blaketingyu committed:
Wrote comments and some organising

blaketingyu committed:
Changed Duke program accordingly to the new Commands class

blaketingyu committed:
Level 6 (No changes)

blaketingyu committed:
Level 6 (No Change)

blaketingyu committed:
Level 6 (No Change)

blaketingyu committed:
Level-6 (Update Commit Message)

blaketingyu committed:
Updated opening message cause previous one was super lame

blaketingyu committed:
Fixed one of the display in print line

blaketingyu committed:
Level-6

blaketingyu committed:
Updated display line when task description is empty

blaketingyu committed:
Updated expected output accordingly to the input file

blaketingyu committed:
Level-7

blaketingyu committed:
Level-8

blaketingyu committed:
Merge branch 'branch-Level-7'

blaketingyu committed:
Solve Merge Conflicts and Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/DeadlineTask.java
#	src/main/java/Parser.java

blaketingyu committed:
Level-9

blaketingyu committed:
Branch to Level-9

blaketingyu committed:
A-MoreOOP

blaketingyu committed:
Merge branch 'branch-A-MoreOOP'

blaketingyu committed:
Add more OOP

blaketingyu committed:
A-Packages
Sort the files into different packages

blaketingyu committed:
Merge branch 'branch-A-Packages'

blaketingyu committed:
A-Jar

Package the App as a Jar File

blaketingyu committed:
A-JUnit

Add JUnit files to test Add,Delete,ParseFiles,Storage and Tasks

blaketingyu committed:
A-JavaDoc

Add Javadocs for all class files

blaketingyu committed:
A-CodingStandard

Revise code for coding standard

blaketingyu committed:
A-CodingStandard

Revise code for coding standard

blaketingyu committed:
Merge branch 'branch-A-JavaDoc'

blaketingyu committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Taskmaster/Commands/Commands.java
#	src/main/java/Taskmaster/Commands/DeleteCommands.java
#	src/main/java/Taskmaster/Exception/DukeExceptions.java
#	src/main/java/Taskmaster/Task/Task.java
#	src/main/java/Taskmaster/Task/TodoTask.java
#	src/main/java/Taskmaster/UserInterface/UserInterface.java
#	src/main/java/Taskmaster/util/TaskList.java

blaketingyu committed:
Change access modifier for fields in task list

blaketingyu committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

blaketingyu committed:
branch-A-Gradle

blaketingyu committed:
branch-A-Gradle

blaketingyu committed:
A-Gradle

blaketingyu committed:
Fix test imports

blaketingyu committed:
Merge branch 'branch-A-Gradle'

blaketingyu committed:
Fix AddCommands out of bound error

blaketingyu committed:
A-CheckStyle

blaketingyu committed:
Refactor code based on checkstyle

blaketingyu committed:
Refactor code based on checkStyle

blaketingyu committed:
Merge branch 'branch-A-CheckStyle'

blaketingyu committed:
Refactor back to start

blaketingyu committed:
Refactor most of the classes file, there are still errors

blaketingyu committed:
Fix find and list commands

blaketingyu committed:
Merge branch 'branch-Level-10'

blaketingyu committed:
Update run path in build.gradle

blaketingyu committed:
Fix GUI Bye Command

blaketingyu committed:
Implement Parser to parse commands

blaketingyu committed:
Fix Display issues

blaketingyu committed:
Add Opening Message

MechFroG88 commented on src/main/java/taskmaster/commands/DeleteCommands.java:
```diff
@@ -0,0 +1,101 @@
+
+            //If index is out of range, throw illegal argument exception
+            if (taskList.isNumberOutOfRange(index)) {
+                throw new DukeExceptions("BRAT ! Your index is out of range! "
```
AHHAHAHA

MechFroG88 commented on src/main/java/taskmaster/commands/DeleteCommands.java:
```diff
@@ -0,0 +1,101 @@
+    private String deleteTask(int index, TaskList taskList) {
+        String result = "";
+        Task selectedTask = taskList.get(index - 1);
+        result += "YES! I've removed this task and soon I'll remove you as well!:\n";
```
I like your chatbot. So polite XDD

MechFroG88 commented on src/main/java/taskmaster/task/EventTask.java:
```diff
@@ -0,0 +1,53 @@
+     *
+     * @param eventDate Time and Date of Event.
+     */
+
```
I think that there shouldn't be a new line here (based on the java docs on module website)

MechFroG88 commented on src/main/java/taskmaster/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+    protected String taskName;
+
+    /** Status of the task. **/
+    protected Boolean isCompleted;
```
I think you should use `boolean` instead of `Boolean` since it is much more efficient

blaketingyu replied:
Thanks for raising this up! Didn't notice I used the wrapper class here.

MechFroG88 reviewed (commented):
Overall pretty nice code!

Just a few suggestion that you can probably take note of.

blaketingyu committed:
Made some changes based on peer review

WJunHong commented on src/main/java/taskmaster/commands/Commands.java:
```diff
@@ -0,0 +1,47 @@
+
+
+    /**
+     * Execute the command.
```
I think there needs to be a new line between the javadoc summary and the param tag

blaketingyu replied:
Thanks for reading through my code and helping me find these mistakes and also giving me suggestions Jun Hong! 💯

WJunHong commented on src/main/java/taskmaster/util/ParseFiles.java:
```diff
@@ -0,0 +1,171 @@
+                //Handle Wrong Inputs
+                //Case 1: user only specified Task type
+                if (stringIntoParts.length == 1) {
+                    throw new DukeExceptions("ERROR! You've only specified the task type, you need to specify"
```
I think can change the name to TaskMasterException

WJunHong commented on src/main/java/taskmaster/exception/DukeExceptions.java:
```diff
@@ -0,0 +1,19 @@
+    /**
+     * Constructor for the DukeExceptions class.
+     *
+     * @param e - Error Message
```
I think can omit the "-"

WJunHong reviewed (commented):
Overall quite good adherence to Java coding standards, good job! :D

blaketingyu committed:
Refactor code based on new parameter in Storage class

Storage class had no parameter.

Having a parameter to specify the file path allows the program to change the file path for the Duke.txt file and /data directory.

blaketingyu committed:
Implement  Assertions in the delete and add commands

The use of assertions helps to ensure that the add commands and delete commands is working as intended.
The size of task list should incremented by one when a task has been added.
The size of task list should be decremented by one when a task has been deleted.

The size of the task list is being compared with it's size after performing the add and delete commands
to ensure that the commands has been successful in executing their respective functions as intended.

Let's add assert statements to each addCommands (Todo, Deadline, Event) and deleteCommands to
ensure that the size of the task list contains one more or one less than the initial size of the task list
after performing the execution.

blaketingyu committed:
Merge branch 'branch-A-Assertions'

blaketingyu committed:
Refactor code based on code quality check
Critically examines the code and refactor to improve the code quality where necessary.

Change variable names if not suitable and created more methods to adhere to the SLAP.

Let's change variable names to improve readability and create more methods to create
more level of abstraction (SLAP).

blaketingyu committed:
Implement SLAP by having loadFile method in Taskmaster

blaketingyu committed:
Merge branch 'branch-A-CodeQuality'

blaketingyu committed:
Add SLAP to have more abstraction, changed variable names to improve readability and overall code quality.

Change variable names to more suitable variable names accordingly to their data type for better readability and overall code quality.
Implement SLAP by having loadFile method to have more abstraction as well.

blaketingyu committed:
Improve GUI by changing the mainwindow and dialogbox FXML files

blaketingyu committed:
Merge branch 'Merging-PRs'

blaketingyu committed:
Improve GUI by changing the mainwindow and dialogbox FXML File

blaketingyu committed:
Revert "Improve GUI by changing the mainwindow and dialogbox FXML File"

This reverts commit d2076f631a43c2b0c4b056a9d2363f0f8a83a079.

blaketingyu committed:
Edit checkstyle.xml file

blaketingyu committed:
Edit checkstyle.xml file

blaketingyu committed:
Edit test file for checkstyle errors

blaketingyu committed:
Test Java CI

blaketingyu committed:
Add text-ui-test to gitignore

blaketingyu committed:
Update gradle.yml

blaketingyu committed:
Implement better search to search for tasks that contains the specified keyword

 Improve the existing search by searching for task names regardless of case sensitivity.

blaketingyu committed:
Improve existing help command.

blaketingyu committed:
Merge branch 'C-BetterSearch'

blaketingyu committed:
Improve existing help command and GUI

blaketingyu committed:
Merge branch 'C-Help'

blaketingyu committed:
Implement TagList to tag the tasks

blaketingyu committed:
Implement A-BetterGui and remove tagging

blaketingyu committed:
Merge branch 'A-BetterGui'

blaketingyu committed:
Update background image

blaketingyu committed:
Improve coding quality based on coding standards

blaketingyu committed:
Add a representative screenshot

blaketingyu committed:
Set theme jekyll-theme-tactile

blaketingyu committed:
Set theme jekyll-theme-midnight

blaketingyu committed:
Update README.md

blaketingyu committed:
Update README.md

blaketingyu committed:
Update README.md

blaketingyu committed:
Update README.md

blaketingyu committed:
Update README.md for user guide

blaketingyu committed:
Set theme jekyll-theme-tactile

blaketingyu committed:
Add screenshots for User Guide

blaketingyu committed:
Merge branch 'master' of https://github.com/blaketingyu/ip

# Conflicts:
#	docs/README.md

blaketingyu committed:
Handle Merge-Conflcit

blaketingyu committed:
Add screenshots to UG

blaketingyu committed:
Rectify issue of data file not being created when the program is run

blaketingyu committed:
Update README.md

blaketingyu committed:
Remove picture for the user guide

blaketingyu committed:
refactor: add system.exit to program when terminating
